pymc_marketing/mmm/transformers.py

```diff
@@ -235,6 +236,11 @@ def geometric_adstock(
        with carryover and shape effects." (2017).
 
     """
+    alpha = check_parameters(
+        alpha, [pt.gt(alpha, 0), pt.lt(alpha, 1)], msg="0 <= alpha <= 1"
+    ).eval()
+    l_max = check_parameters(l_max, [pt.gt(l_max, 0)], msg="l_max >= 0").eval()
```

Comment:
We won't want to use eval here

Reply:
Yep, I am seeing issues. 

Removing and having a look.

Reply:
It should only raise when the eval is called on the result of function

```python
# Fine
result = geometric_adstock(np.linspace(0, 1), alpha=-5)
# Causes error
result.eval()

Reply:
Just like the PyMC distributions: 

```python
# Fine
dist = pm.Normal.dist(mu=0, sigma=-1)
# ValueError
dist.eval()
```

Reply:
Can this check not all be done in the same line? like in the example I linked?

Reply:
I removed the check on l_max > 0:
* If passed, an exception is already raised: `ValueError: length should not be negative`
* `batched_convolution` is called ahead of `.eval()`. So the previous exception is already raised and we never reach `ParameterValueError`

Reply:
`.eval()` now part of the test as: 

```
y = geometric_adstock(x=x, alpha=alpha, l_max=l_max)
y.eval()
```

Reply:
Perf. Don't think we are passing l_max as TensorVariable anyway